echopype/echodata/combine.py

```diff
@@ -0,0 +1,280 @@
+        starting at the timestamp where the reversal occurs such that all
+        `ping_time`s and `location_time`s in the output are monotonically increasing.
+        Additionally, the original `ping_time` and `location_time` values will be stored
+        in the provenance group, although this behavior may change in future versions.
+
+        Attributes from all groups before the combination will be stored in the provenance group,
+        although this behavior may change in future versions.
+
+        The `source_file` and `converted_raw_path` attributes will be copied from the first
+        `EchoData` object in the given list, but this may change in future versions.
+
+    Raises
+    ------
+    ValueError
+        If `echodatas` contains `EchoData` objects with different or `None` `sonar_model` values
+        (i.e., all `EchoData` objects must have the same non-None `sonar_model` value).
+    ValueError
+        If EchoData objects have conflicting source file names.
+
+    Warns
+    -----
+    UserWarning
+        If the `sonar_model` of the input `EchoData` objects is `"EK60"` and any `EchoData` objects
+        have non-monotonically increasing `ping_time`s or `location_time`s,
+        the `ping_time`s or `location_time`sin the output `EchoData` object will be increased
+        starting at the timestamp where the reversal occurs such that all
+        `ping_time`s and `location_time`s in the output are monotonically increasing.
+
+    Warnings
+    --------
+    Changes in parameters between `EchoData` objects are not currently checked;
+    however, they may raise an error in future versions.
+
+    Notes
+    -----
+    `EchoData` objects are combined by combining their groups individually.
+
+    Examples
+    --------
+    >>> ed1 = echopype.open_converted("file1")
+    >>> ed2 = echopype.open_converted("file2")
+    >>> combined = echopype.combine_echodata([ed1, ed2])
+    """
+
+    result = EchoData()
+    if len(echodatas) == 0:
+        return result
+    result.source_file = echodatas[0].source_file
+    result.converted_raw_path = echodatas[0].converted_raw_path
+
+    sonar_model = None
+    for echodata in echodatas:
+        if echodata.sonar_model is None:
+            raise ValueError(
+                "all EchoData objects must have non-None sonar_model values"
+            )
+        elif sonar_model is None:
+            sonar_model = echodata.sonar_model
+        elif echodata.sonar_model != sonar_model:
+            raise ValueError(
+                "all EchoData objects must have the same sonar_model value"
+            )
+
+    # ping time before reversal correction
+    old_ping_time = None
+    # ping time after reversal correction
+    new_ping_time = None
+    # location time before reversal correction
+    old_location_time = None
+    # location time after reversal correction
+    new_location_time = None
+
```

Comment:
So, there are other "time" variables:
https://github.com/OSOceanAcoustics/echopype/blob/e01d57592385324b5747fb5c98d498ff50c42194/echopype/convert/set_groups_base.py#L25-L34

I was wondering if these are also handled? Not sure if like `mru_time` would have reversal or not, or the other `ping_time*` If so, then this would miss it, because it seems like only `ping_time` and `location_time` are handled.

Reply:
yes the `mru_time` needs to be added